lib/src/ioctl/g_ext_ctrls.rs

```diff
@@ -0,0 +1,418 @@
+    }
+}
+
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, PartialEq, N)]
+pub enum YCbCrEncoding {
+    Default = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_DEFAULT,
+    E601 = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_601,
+    E709 = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_709,
+    Xv601 = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_XV601,
+    Xv709 = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_XV709,
+    Sycc = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_SYCC,
+    Bt2020 = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_BT2020,
+    Bt2020ConstLum = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_BT2020_CONST_LUM,
+    Smpte240M = bindings::v4l2_ycbcr_encoding_V4L2_YCBCR_ENC_SMPTE240M,
+}
+
+impl TryFrom<u32> for YCbCrEncoding {
+    type Error = DataConversionError;
+
+    fn try_from(ycbcr_encoding: u32) -> Result<Self, Self::Error> {
+        YCbCrEncoding::n(ycbcr_encoding).ok_or(Self::Error::YCbCrEncodingErr)
+    }
+}
+
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, PartialEq, N)]
+pub enum Quantization {
+    Default = bindings::v4l2_quantization_V4L2_QUANTIZATION_DEFAULT,
+    FullRange = bindings::v4l2_quantization_V4L2_QUANTIZATION_FULL_RANGE,
+    LimRange = bindings::v4l2_quantization_V4L2_QUANTIZATION_LIM_RANGE,
+}
+
+impl TryFrom<u32> for Quantization {
+    type Error = DataConversionError;
+
+    fn try_from(quantization: u32) -> Result<Self, Self::Error> {
+        Quantization::n(quantization).ok_or(Self::Error::QuantizationErr)
+    }
+}
+
+bitflags! {
+    /// FWHT Flags.
+    pub struct FwhtFlags: u32 {
+        const INTERLACED = bindings::V4L2_FWHT_FL_IS_INTERLACED as u32;
+        const BOTTOM_FIRST = bindings::V4L2_FWHT_FL_IS_BOTTOM_FIRST as u32;
+        const ALTERNATE = bindings::V4L2_FWHT_FL_IS_ALTERNATE as u32;
+        const BOTTOM_FIELD = bindings::V4L2_FWHT_FL_IS_BOTTOM_FIELD as u32;
+        const UNCOMPRESSED = bindings::V4L2_FWHT_FL_LUMA_IS_UNCOMPRESSED as u32;
+        const CB_COMPRESSED = bindings::V4L2_FWHT_FL_CB_IS_UNCOMPRESSED as u32;
+        const CR_COMPRESSED = bindings::V4L2_FWHT_FL_CR_IS_UNCOMPRESSED as u32;
+        const CHROMA_FULL_HEIGHT = bindings::V4L2_FWHT_FL_CHROMA_FULL_HEIGHT as u32;
+        const CHROMA_FULL_WIDTH = bindings::V4L2_FWHT_FL_CHROMA_FULL_WIDTH as u32;
+        const ALPHA_UNCOMPRESSED = bindings::V4L2_FWHT_FL_ALPHA_IS_UNCOMPRESSED as u32;
+        const I_FRAME = bindings::V4L2_FWHT_FL_I_FRAME as u32;
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct FwhtParams {
+    pub backward_ref_ts: u64,
+    pub version: u32,
+    pub width: u32,
+    pub height: u32,
+    pub flags: u32,
+    pub components: u32,
+    pub encoding: u32,
+    pub colorspace: Colorspace,
+    pub xfer_func: XferFunc,
+    pub ycbcr_enc: YCbCrEncoding,
+    pub quantization: Quantization,
+}
+
+// Private enum serving as storage for control-specific bindgen struct
+#[allow(clippy::large_enum_variant)]
+enum CompoundControl {
```

Comment:
Doesn't this need to be public? `cargo clippy` reports that no member but `FwhtParams` are constructed, which means the rest is dead code with the current visibility.

Reply:
As things stand now, it probably shouldn't be public for the reasons set forth in the comment. It is true that H264 stuff is unused at the moment. We expect quite a lot of stuff here for other codecs and I wanted to see how it works out. So there are two options: remove H264 for now or complete its support before this PR is actually ready for merging. I'm personally in favour of the former in order to start small, but up to you.